File: backend/src/modules/accounts/domain/repositories/account.repository.ts
import { Account, AccountType } from '../entities/account.entity';

export abstract class AccountRepository {
  /** All non-archived-in-DB (not soft-deleted) accounts owned by the user. */
  abstract findAllForUser(userId: string): Promise<Account[]>;

  abstract findById(id: string): Promise<Account | null>;

  abstract create(data: {
    userId: string;
    type: AccountType;
    name: string;
    currency: string;
  }): Promise<Account>;

  abstract update(
    id: string,
    changes: { name?: string; archived?: boolean },
  ): Promise<Account>;

  abstract softDelete(id: string): Promise<void>;

  /** Persists a recomputed balance_cached — distinct from update() since
   * it's never driven by the PATCH DTO (name/archived), only by
   * RecalculateAccountBalanceService. */
  abstract setBalance(id: string, balance: string): Promise<void>;
}

File: mobile/lib/features/wallet/domain/repositories/account_repository.dart
import '../entities/account.dart';

abstract class AccountRepository {
  /// GET /accounts — the caller's own, non-deleted accounts.
  Future<List<Account>> fetchAll();

  /// PATCH /accounts/{id} — rename/archive only (docs/08_API.md §8).
  Future<Account> update(String id, {String? name, bool? archived});
}
